baseapp-chats/baseapp_chats/graphql/object_types.py

```diff
@@ -111,17 +116,37 @@ def get_node(cls, info, id):
         except cls._meta.model.DoesNotExist:
             return None
 
+    def resolve_all_messages(self, info, **kwargs):
+        return self.messages.all().order_by("-created")
+
+    def resolve_is_archived_by_current_profile(self, info, **kwargs):
+        profile_pk = (
+            info.context.user.current_profile.pk
+            if hasattr(info.context.user, "current_profile")
+            and hasattr(info.context.user.current_profile, "pk")
+            else None
+        )
+        participant = self.participants.filter(profile_id=profile_pk).first()
```

Comment:
we can do this instead, it avoids instantiating a model instance, we just need to know if its true or false :) 
```suggestion
        return self.participants.filter(profile_id=profile_pk, has_archived_room=True).exists()
```